### PR TITLE
testing: refactor release helper and switch flags to commands

### PR DESCRIPTION
**Pull request by eliben**
Refactoring prepares for adding new commands, such as bumping versions
for release testing.

### CODE REVIEW

File: internal/testing/releasehelper.go [hunk 1]
```diff
@@ -132,24 +153,32 @@ func runOnGomod(path string) {
...
-			} else if *doDropReplace {
-				cmdCheck(fmt.Sprintf("go mod edit -dropreplace=%s %s", r.Path, gomodPath))
-			}
+			reqHandler(gomodPath, r.Path, rel)
 		}
 	}
 }
 
 func main() {
-	flag.Usage = func() {
```
Comment (vangent) [inline internal/testing/releasehelper.go]: Consider using `github.com/google/subcommands`. It does a better job at `help` than you, and will set up the commands as struct that can define their own flags, etc.

https://github.com/google/go-cloud/blob/master/samples/gocdk-secrets/main.go has a simple example.

Comment (eliben) [reply internal/testing/releasehelper.go]: I considered it, and I'd rather not. Right now these simple helper tools are single-file. If they grow dependencies they'll need to be split into separate modules (or some unified `tools` module). 

Moreover, it's not clear to me what the advantage of `subcommands` is for this specific tool. It seems pretty trivial to me :man_shrugging:  Commands can have their own flags with `flag.NewFlagSet` if needed.

Besides, we use cobra for the gocdk which makes sense, but how many different subcommand packages we really need in a single project? I'd vote to only use cobra for complicated stuff like the CLI and the standard library for everything else.

File: internal/testing/releasehelper.go [hunk 2]
```diff
@@ -39,16 +39,27 @@ The tool processes all modules listed in the 'allmodules'
...
+
+  addreplace    adds 'replace' directives to point to local versions
+                for testing.
 
-Usage:
+  dropreplace   removes these directives.
+
+  help          prints this usage message
+
+Run it from the root directory of the project.
```
Comment (vangent) [inline internal/testing/releasehelper.go]: project -> repository ?

File: internal/testing/releasehelper.go [hunk 3]
```diff
@@ -39,16 +39,27 @@ The tool processes all modules listed in the 'allmodules'
...
 `
 
-var doReplace = flag.Bool("replace", false, "add replace directives")
-var doDropReplace = flag.Bool("dropreplace", false, "drop replace directives")
+func printHelp() {
+	_, binName := filepath.Split(os.Args[0])
```
Comment (vangent) [inline internal/testing/releasehelper.go]: Sorry, what does this do? Why not just `fmt.Println(helpText)`? And maybe `fmt.Fprintf(os.Stderr` ?

File: internal/testing/releasehelper.go [hunk 4]
```diff
@@ -132,24 +153,32 @@ func runOnGomod(path string) {
...
+	case "addreplace":
+		handlerFunc = func(gomodPath string, mod string, modPath string) {
+			cmdCheck(fmt.Sprintf("go mod edit -replace=%s=%s %s", mod, modPath, gomodPath))
+		}
+	case "dropreplace":
+		handlerFunc = func(gomodPath string, mod string, modPath string) {
+			cmdCheck(fmt.Sprintf("go mod edit -dropreplace=%s %s", mod, gomodPath))
+		}
+	default:
+		log.Fatal("Expected 'addreplace' or 'dropreplace'")
```
Comment (vangent) [inline internal/testing/releasehelper.go]: Print usage here and exit(1)?

File: internal/testing/releasehelper.go [hunk 5]
```diff
@@ -132,24 +153,32 @@ func runOnGomod(path string) {
...
-	flag.Parse()
-	if *doReplace && *doDropReplace {
-		log.Fatal("Expected only one of -replace/-dropreplace")
+	var handlerFunc reqHandlerFunc
+	switch os.Args[1] {
+	case "help":
+		printHelp()
+		os.Exit(0)
+	case "addreplace":
+		handlerFunc = func(gomodPath string, mod string, modPath string) {
```
Comment (vangent) [inline internal/testing/releasehelper.go]: Nit: `func(gomodPath, mod, modPath string)` here and below.

File: internal/testing/releasehelper.go (hunk 2 above)
Comment (eliben) [reply internal/testing/releasehelper.go]: Done

File: internal/testing/releasehelper.go (hunk 3 above)
Comment (eliben) [reply internal/testing/releasehelper.go]: Without this the command is shown as `/tmp/go-build736844560/b001/exe/releasehelper` when running with `go run`.

Just figured it's nice to have the command name there for usage.

File: internal/testing/releasehelper.go (hunk 4 above)
Comment (eliben) [reply internal/testing/releasehelper.go]: Done

File: internal/testing/releasehelper.go (hunk 5 above)
Comment (eliben) [reply internal/testing/releasehelper.go]: Done

File: internal/testing/releasehelper.go (hunk 3 above)
Comment (vangent) [reply internal/testing/releasehelper.go]: Ah, I see, I missed that `helpText` had a '%s' in it.

**Pull request merged**